core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -135,9 +139,27 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
         }
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
+        builder.field(STATUS, status.getStatus());
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+
+        String currentFieldName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            }
```

Comment:
i don't think you need to save the currentName to a variable here, this can be a single `if (token == FIELD_NAME && STATUS.equals(parser.currentName()) == false)`